packages/aws-appsync-subscription-link/src/realtime-subscription-handshake-link.ts

```diff
@@ -696,6 +696,10 @@ export class AppSyncRealTimeSubscriptionHandshakeLink extends ApolloLink {
 
   private _timeoutDisconnect() {
     this.subscriptionObserverMap.forEach(({ observer }) => {
+      if (!observer) {
+        return;
+      }
+
```

Comment:
@KvNGCzA I think this should never happen. If the map is empty, the callback wouldnt been call? Did you experience something that the observer is `undefined`. I think the issue is somewhere else. I will keep looking

Reply:
@elorzafe After a subscription is established, I randomly get the following error(s): `Undefined is not an object evaluating (_a.observer)` or `Cannot read property 'observer' of undefined`

These errors appear randomly in my case. I'm suspecting it has to do with something with the subscription timing out.

Reply:
I was able to reproduce this error by randomly switching between networks(WiFi/4G). I have already rollout a build with this patch and I am no longer getting this crash. Thank you